syntax = "proto3";

service TaskService {
  rpc SendTask(TaskRequest) returns (Ack);
  rpc GetQueueLength(Empty) returns (QueueLengthResponse);
  rpc GetCPUUsage(Empty) returns (CPUUsageResponse);
  rpc AnnouncePresence(ServerInfo) returns (Ack);
}

message Empty {}

message TaskRequest {
  string name = 1;
  int32 weight = 2;
  bool replicated = 3;
  int32 source_id = 4;
}

message Ack {
  bool success = 1;
}

message QueueLengthResponse {
  int32 length = 1;
}

message CPUUsageResponse {
  double usage = 1;
}

message ServerInfo {
  int32 server_id = 1;
  string host = 2;
  int32 dist = 3;
}
